delphi-frontend/src/main/antlr3/au/com/integradev/delphi/antlr/Delphi.g

```diff
@@ -1182,6 +1187,12 @@ simpleNameReference          : ident
 extendedNameReference        : extendedIdent genericArguments? ('.' extendedNameReference)?
                              -> ^(TkNameReference<NameReferenceNodeImpl> extendedIdent genericArguments? ('.' extendedNameReference)?)
                              ;
+routineNameReference         : ident genericDeclarationArguments? ('.' extendedRoutineNameReference)?
+                             -> ^(TkNameReference<NameReferenceNodeImpl> ident genericDeclarationArguments? ('.' extendedRoutineNameReference)?)
+                             ;
+extendedRoutineNameReference : extendedIdent genericDeclarationArguments? ('.' extendedRoutineNameReference)?
```

Comment:
I think `routineNameReference` and `extendedRoutineNameReference` are also somewhat misleading, since they are only used in the very specific circumstance of a routine implementation heading. Not sure offhand what we might call this instead... any thoughts?

Reply:
I don't love the idea of putting any more information in these names, as `extendedRoutineNameReference` is already exactly long enough to avoid breaking the formatting in the grammar file.
But if we were to do this, we'd probably just be sticking the word `Implementation` in there.